internal/hcl/constraints.go

```diff
@@ -124,47 +125,53 @@ func (p *ProviderConstraints) MarshalJSON() ([]byte, error) {
 // struct from the required_providers configuration. Currently, we only support
 // AWS provider constraints.
 func NewProviderConstraints(blocks Blocks) *ProviderConstraints {
+	var allConstraints ProviderConstraints
 	terraformBlocks := blocks.OfType("terraform")
 	for _, block := range terraformBlocks {
 		req := block.GetChildBlock("required_providers")
 		if req == nil {
 			continue
 		}
 
-		def := req.Values().AsValueMap()
-		for provider, body := range def {
-			if provider != "aws" {
-				continue
-			}
+		if constraints, err := readProviderConstraints(req, "aws"); err == nil {
+			allConstraints.AWS = constraints
+		}
+		if constraints, err := readProviderConstraints(req, "google"); err == nil {
+			allConstraints.Google = constraints
+		}
+	}
+	return &allConstraints
+}
 
-			if body.IsNull() || !body.IsKnown() {
-				continue
-			}
+func readProviderConstraints(requiredProviderBlock *Block, provider string) (version.Constraints, error) {
```

Comment:
Would it be worth adding a test for this?

Reply:
Yes